ArduPlane/tailsitter.cpp

```diff
@@ -53,20 +53,73 @@ void QuadPlane::tailsitter_output(void)
         return;
     }
 
-    float tilt_left = 0.0f;
-    float tilt_right = 0.0f;
-
-    if (!tailsitter_active() || in_tailsitter_vtol_transition()) {
-        if (tailsitter.vectored_forward_gain > 0) {
-            // thrust vectoring in fixed wing flight
-            float aileron = SRV_Channels::get_output_scaled(SRV_Channel::k_aileron);
-            float elevator = SRV_Channels::get_output_scaled(SRV_Channel::k_elevator);
-            tilt_left  = (elevator + aileron) * tailsitter.vectored_forward_gain;
-            tilt_right = (elevator - aileron) * tailsitter.vectored_forward_gain;
+    // record plane gain outputs before overwriting with VTOL gains in case they are needed for interpolation
+    float fw_aileron;
+    float fw_elevator;
+    float fw_rudder;
+    float fw_tilt_left = 0;
+    float fw_tilt_right = 0;
+    //float fw_throttle_left;
+    //float fw_throttle_right;
+
+    if (tailsitter_active()) { // if in Qmode we need to match the Q rates with plane controller
+        // no fixed wing yaw controller so cannot stabilize VTOL roll
+        //const float roll_rate = attitude_control->get_rate_roll_pid().get_pid_info().desired * 100;
+        const float pitch_rate = attitude_control->get_rate_pitch_pid().get_pid_info().desired * 100;
+        const float yaw_rate = attitude_control->get_rate_yaw_pid().get_pid_info().desired * 100;
+        const float speed_scaler = plane.get_speed_scaler();
+
+        // due to reference frame change roll and yaw are swapped
+        fw_aileron = plane.rollController.get_rate_out(-yaw_rate,  speed_scaler);
+        fw_elevator = plane.pitchController.get_rate_out(pitch_rate, speed_scaler);
+        //fw_rudder = plane.yawController.get_rate_out(roll_rate, speed_scaler);
+
+        // use roll as rudder input and output direct as with plane
+        fw_rudder = plane.channel_roll->get_control_in();
+
+        /* Use VTOL throttle as no rudder stabilization in plane yet
+        const float throttle = motors->get_throttle() * 100;
+        const float rud_gain = float(plane.g2.rudd_dt_gain) / 100;
+
+        // Calculate differential thrust in forward flight
```

Comment:
why is all of this commented out?

Reply:
originally it was there so it could be un-commented when plane got a yaw rate controller. I'm in the process of a rebase and tidy up.